test/vvenc_unit_test/vvenc_unit_test.cpp

```diff
@@ -454,6 +462,80 @@ static bool test_InterPred()
 }
 #endif
 
+#if ENABLE_SIMD_OPT_DIST
+static bool check_lumaWeightedSSE( RdCost* ref, RdCost* opt, unsigned num_cases, int width, int height )
+{
+  printf( "Testing RdCost::lumaWeightedSSE %dx%d\n", width, height );
+
+  std::ostringstream sstm;
+  sstm << "lumaWeightedSSE" << " w=" << width << " h=" << height;
+
+  bool passed = true;
+  for( unsigned i = 0; i < num_cases; i++ )
+  {
+    DimensionGenerator rng;
+    int stride = rng.get( 1, 1024 );
+    int size = std::max( stride, width );
+    std::vector<Pel> orgBuf( size * height );
+    std::vector<Pel> curBuf( size * height );
+    std::vector<Pel> orgLumaBuf( size * height * 4 );
+    std::vector<uint32_t> lumaWeights( 1024 );
+
+    DistParam dtParam;
+    dtParam.org.buf = orgBuf.data();
+    dtParam.cur.buf = curBuf.data();
+    dtParam.org.width = width;
+    dtParam.org.height = height;
+    dtParam.cur.stride = stride;
+    dtParam.org.stride = stride;
+    CPelBuf pelBuf;
+    pelBuf.buf = orgLumaBuf.data();
+    pelBuf.stride = stride;
+    dtParam.orgLuma = &pelBuf;
+    dtParam.bitDepth = 10;
+    dtParam.compID = i % 2 ? COMP_Y : COMP_Cb;
```

Comment:
If `num_cases == 1` no chroma test is performed. I think `num_cases` should not be used to control how to perform a test. But than, the tested `lumaWeightedSSE`, might not need be tested for chroma.

Reply:
I am sorry, I think I was not too clear. Thanks you for separating num_cases from deciding on what to test. But since the function tested is only meant for luma signals, the following can always be used: `dtPAram.compID = COMP_Y`